internal/sbi/server.go

```diff
@@ -0,0 +1,151 @@
+package sbi
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"runtime/debug"
+	"sync"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/sirupsen/logrus"
+
+	"github.com/free5gc/amf/internal/logger"
+	"github.com/free5gc/amf/internal/sbi/consumer"
+	"github.com/free5gc/amf/internal/sbi/processor"
+	"github.com/free5gc/amf/pkg/app"
+	"github.com/free5gc/amf/pkg/factory"
+	"github.com/free5gc/openapi/models"
+	"github.com/free5gc/util/httpwrapper"
+	logger_util "github.com/free5gc/util/logger"
+)
+
+type ServerAmf interface {
+	app.App
+
+	Consumer() *consumer.Consumer
+	Processor() *processor.Processor
+}
+
+type Server struct {
+	ServerAmf
+
+	httpServer *http.Server
+	router     *gin.Engine
+}
+
+func NewServer(amf ServerAmf, tlsKeyLogPath string) (*Server, error) {
+	s := &Server{
+		ServerAmf: amf,
+	}
+
+	s.router = newRouter(s)
+
+	cfg := s.Config()
+	bindAddr := cfg.GetSbiBindingAddr()
+	logger.SBILog.Infof("Binding addr: [%s]", bindAddr)
+	var err error
+	if s.httpServer, err = httpwrapper.NewHttp2Server(bindAddr, tlsKeyLogPath, s.router); err != nil {
+		logger.InitLog.Errorf("Initialize HTTP server failed: %v", err)
+		return nil, err
+	}
+	s.httpServer.ErrorLog = log.New(logger.SBILog.WriterLevel(logrus.ErrorLevel), "HTTP2: ", 0)
+
+	return s, nil
+}
+
+func newRouter(s *Server) *gin.Engine {
+	router := logger_util.NewGinWithLogrus(logger.GinLog)
+
+	amfCommunicationGroup := router.Group(factory.AmfCommResUriPrefix)
+	amfCommunicationRoutes := s.getCommunicationRoutes()
+	applyRoutes(amfCommunicationGroup, amfCommunicationRoutes)
+
+	amfEventExposureGroup := router.Group(factory.AmfEvtsResUriPrefix)
+	amfEventExposureRoutes := s.getEventexposureRoutes()
+	applyRoutes(amfEventExposureGroup, amfEventExposureRoutes)
+
+	amfHttpCallBackGroup := router.Group(factory.AmfCallbackResUriPrefix)
+	amfHttpCallBackRoutes := s.getHttpCallBackRoutes()
+	applyRoutes(amfHttpCallBackGroup, amfHttpCallBackRoutes)
+
+	amfLocationGroup := router.Group(factory.AmfLocResUriPrefix)
+	amfLocationRoutes := s.getLocationRoutes()
+	applyRoutes(amfLocationGroup, amfLocationRoutes)
+
+	amfMTGroup := router.Group(factory.AmfMtResUriPrefix)
+	amfMTRoutes := s.getMTRoutes()
+	applyRoutes(amfMTGroup, amfMTRoutes)
+
+	amfOAMGroup := router.Group(factory.AmfOamResUriPrefix)
+	amfOAMRoutes := s.getOAMRoutes()
+	applyRoutes(amfOAMGroup, amfOAMRoutes)
+
+	return router
+}
+
+func (s *Server) Run(traceCtx context.Context, wg *sync.WaitGroup) error {
+	var profile models.NfProfile
+	if profileTmp, err1 := consumer.GetConsumer().BuildNFInstance(s.Context()); err1 != nil {
```

Comment:
Hi @donald1218 @tim1207 and @andy89923 
I think all of the use of `consumer.GetConsumer()` in `sbi` package can use `s.Consumer()` instead.

Reply:
please also remember to remove commented lines if they are not in use anymore.

Tks!